Updated support for angular rc.6

OnlyAGhost opened the pull request:
Had to add a module for angular2-grid, replaced calls from depricated DynamicComponentLoader to componentFactoryResolver. To use in rc6 and beyond, in app NgModule import NgGridModule and the rest should load. Demo or the index under src are not updated with the new calls.

mlittrell commented:
+1

jvendryes commented:
What @mlittrell said is correct, those are required versions now: https://github.com/angular/angular/blob/master/CHANGELOG.md#peer-dependency-updates

OnlyAGhost commented:
I went ahead and updated with the latest version recommended version from angular of zone and rxjs. I added this into the support ticket as well, but figure this gives others a way to look at it:

---
FYI, I updated my fork with some new code. Fixed the calls to index.html, had to fix some things in the module (needed to include FormsModule), just generally made it more in-line with the recent Angular changes. I also added lite-server, so in npm you can do npm run lite and it'll automatically start a webserver for the project. I tested running http://localhost:3000/src/index.html and it correctly shows the grid, allows you to add/remove/drag, etc.
---
Demo site still needs to be updated with that code but everything else seems to be checking out.

BTMorton commented on package.json:
```diff
@@ -41,18 +43,24 @@
+    "@angular/forms": "2.0.0-rc.6",
+    "@angular/platform-browser": "2.0.0-rc.6",
+    "@angular/platform-browser-dynamic": "2.0.0-rc.6",
+    "angular2-in-memory-web-api": "0.0.18",
```
What's this included for?

BTMorton commented on src/index.html:
```diff
@@ -2,34 +2,19 @@
-		<script src="node_modules/reflect-metadata/Reflect.js"></script>
-		<script src="node_modules/systemjs/dist/system.js"></script>
+		<link rel="stylesheet" href="/node_modules/bootstrap/dist/css/bootstrap.min.css" />
+		<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
```
Why have you included jquery?

OnlyAGhost replied:
The original index.html was calling in bower_components/bootstrap which was probably copied over from another source tree. I was pulling in the bootstrap library for the css which was already in the index and instead of including the whole jquery library just referenced it from google directly.

BTMorton replied:
Bootstrap is included in the bower dependencies. There's no harm moving to npm as none of the other deps are in bower. However, it's just the CSS imported from bootstrap. None of the js is used so jquery isn't required :)

BTMorton commented on src/modules/NgGrid.module.ts:
```diff
@@ -0,0 +1,13 @@
+import { NgGrid, NgGridItem, NgGridItemConfig, NgGridItemEvent, NgGridPlaceholder } from '../main';
+
+@NgModule({
+  imports:          [ BrowserModule, CommonModule, FormsModule ],
```
Do we need to import all of these? The forms module, for example, isn't used by the library

OnlyAGhost replied:
Maybe I placed some of those into the wrong type of dependency. In the app.html which is in the src directory for what I assume is testing which has ngModule binds which now require the @angular/forms module.

OnlyAGhost commented:
So first (and this would be your call) I removed the ^ to peg angular2-grid against a specific version of @angular repository until their "release candidates" stop breaking everything each release. 

In terms of the libraries, you need common, compiler is used which replaced the DynamicComponentLoader, core is core, forms is required for doing any ngModule binding now, platform-browser and platform-browser-dynamic are needed for the BrowserModule, and I'm not really sure on the in-memory-web-api if it's needed or not. Angular folks always include it but I'm not sure.

BTMorton commented:
Thanks for this. There are a lot of stylistic changes which I'm not overly keen on. I'm imagining they were automatically applied by your editor? I'll try and pull your repo to test it all out ASAP, but I don't have much free time at the moment. Apologies for the delay on this one.

BTMorton commented:
It was the in memory API specifically that I was questioning. The other changes in that file are fine :) I'll look into that one

OnlyAGhost commented:
Yeah pull anything out that doesnt make any sense! Like I said, I've not done much on GitHub so there is definitely some things that I'm still picking up on. In terms of that library here is the details:
https://github.com/angular/in-memory-web-api

It looks like it's used in the new angular router which shouldnt apply here. For changes like that do you want me to rip that out and push and update to my fork or do you make that in merging?

OnlyAGhost commented:
In terms of the stylisting changes, yeah I use VS Code with a linter. Since there isnt a linter config I wouldnt have thought it would have went crazy with it, but there are definitely some changes that it prefers. For instance favoring single quotes instead of double quotes.

BTMorton commented:
That's fine. I tend to switch between the two anyway. Should probably add a lint config...

The in memory API thing is for demo setups. It simulates a web server API for testing http requests as best I can tell. Shouldn't be required here.

Whether you make the changes or I do doesn't matter overly much. You'll probably be slightly faster than I getting to it though :P

Thanks for this. I appreciate the help

OnlyAGhost commented:
I just pushed out some new changes based on that feedback. Removed jquery reference (which originally I added because I had also added the script for bootstrap, but pulled that script without pulling the jquery script reference for whatever reason). I removed angular2-in-memory-web-api from systemjs.config.js and from packages.json, and removed FormsModule from the NgGrid module which was properly located in the app.module instead.

Should the css for bootstrap just be pulled from bootstrap directly to remove it from being a requirement at all?

BTMorton commented:
One things I've just found from trying your branch, the systemjs config doesn't move to the dist folder when you run the build.

OnlyAGhost commented:
Do you need the systemjs config in dist? I think you only need that for the index.html file. People should have their own systemjs config for their project.

Sent from my iPhone

On Sep 13, 2016, at 5:28 PM, Ben Morton <notifications@github.com<mailto:notifications@github.com>> wrote:


One things I've just found from trying your branch, the systemjs config doesn't move to the dist folder when you run the build.

-
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/BTMorton/angular2-grid/pull/136#issuecomment-246830677>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AF4yzVEfM1_5Q1FzJbTW2RijsI5vMIM7ks5qpxWEgaJpZM4J3dG5>.

BTMorton commented:
The dist for this project is the folder that the included demo app should be run from. It won't be included with the package as it's not listed in the package.json includes. Similarly, the lite-server should probably have a config to set that as it's root directory. I've got some time now so hopefully I can get this running, merged and published this evening

OnlyAGhost commented:
Awesome sounds good! Like I said I'm not super familiar so I though the whole dist was packaged up. That makes a lot more sense! Let me know if you need me to make any changes.

Sent from my iPhone

On Sep 13, 2016, at 5:38 PM, Ben Morton <notifications@github.com<mailto:notifications@github.com>> wrote:


The dist for this project is the folder that the included demo app should be run from. It won't be included with the package as it's not listed in the package.json includes. Similarly, the lite-server should probably have a config to set that as it's root directory. I've got some time now so hopefully I can get this running, merged and published this evening

-
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/BTMorton/angular2-grid/pull/136#issuecomment-246833321>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AF4yzRTD0nhDdrrkEycq5vY7-qlptbFlks5qpxe9gaJpZM4J3dG5>.
